src/curry.ts

```diff
@@ -91,21 +91,51 @@ export const memo = <TFunc extends Function>(
   return memoize({}, func as any, key, ttl) as any as TFunc
 }
 
+export type DebounceFunction<TArgs extends any[]> = {
+  (...args: TArgs): void;
+  /**
+   * Cancels the debounced function
+   */
+  cancel(): void;
+  /**
+   * Runs the debounced function immediately
+   */
+  flush(...args: TArgs): void;
+}
+
 /**
  * Given a delay and a function returns a new function
  * that will only call the source function after delay
- * milliseconds have passed without any invocations
+ * milliseconds have passed without any invocations.
+ *  
+ * The debounce function comes with a `cancel` method 
+ * to cancel delayed `func` invocations and a `flush` 
+ * method to invoke them immediately
  */
 export const debounce = <TArgs extends any[]>(
   { delay }: { delay: number },
   func: (...args: TArgs) => any
-): ((...args: TArgs) => void) => {
-  let timer: any = null
-  const debounced = (...args: TArgs) => {
-    clearTimeout(timer)
-    timer = setTimeout(() => func(...args), delay)
+) => {
+  let timer: NodeJS.Timeout = null
+  let debouncedActive = true 
```

Comment:
Because the context is so small and clear, I would support calling this var `active`